Sources/StreamChat/Controllers/ListDatabaseObserver.swift

```diff
@@ -172,7 +172,11 @@ class ListDatabaseObserver<Item, DTO: NSManagedObject> {
                 // `itemCreator` returns non-optional value, so we can use implicitly unwrapped optional
                 var result: Item!
                 context.performAndWait {
-                    result = itemCreator(dto)
+                    do {
+                        result = try itemCreator(dto)
+                    } catch {
+                        log.assertionFailure("Unable to convert a DB entity to model")
+                    }
```

Comment:
Won't this cause a crash anyway? `result` is still `nil` if error is thrown

Reply:
Good point. Adding it